### DISCUSSION TITLE
Q. Portability : AppImage vs Flatpak?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AppImage/AppImageKit#1239] [Q&A]
I am wondering how AppImage ensures portability and, because I've been experimenting with Flatpak, I am comparing the two in this regard.

As I understand it, each Flatpak carries with it its own "runtime" -- a container -- an isolated system which contains everything the app needs.

But how does AppImage work? How do we guarantee that it will run on any system, or at least, on most systems.

As I understand, new AppImages should be tested against Ubuntu 18 LTS. Is there anything else?

**Comment by Drsheppard01** [comment]
This is most often due to the version of the glibc library, for example, the VSCodium project is built on Glibc version 2.17. That is potentially AppImage can run on all distributions from Ubuntu 13.04, RHEL 7, Fedora 19
